typespec-extension/src/code-model-builder.ts

```diff
@@ -564,108 +517,146 @@ export class CodeModelBuilder {
         codeModelClient.globalParameters!,
         codeModelClient.apiVersions,
       );
-      clientContext.preProcessOperations(this.sdkContext, client);
-
-      const operationGroups = listOperationGroups(this.sdkContext, client, true);
 
-      const operationWithoutGroup = listOperationsInOperationGroup(this.sdkContext, client);
+      // preprocess operation groups and operations
+      // operations without operation group
+      const serviceMethodsWithoutSubClient = this.listServiceMethodsUnderClient(client);
       let codeModelGroup = new OperationGroup("");
-      for (const operation of operationWithoutGroup) {
-        if (!this.needToSkipProcessingOperation(operation, clientContext)) {
-          codeModelGroup.addOperation(this.processOperation("", operation, clientContext));
+      for (const serviceMethod of serviceMethodsWithoutSubClient) {
+        if (!this.needToSkipProcessingOperation(serviceMethod.__raw, clientContext)) {
+          codeModelGroup.addOperation(this.processOperation(serviceMethod, clientContext, ""));
         }
       }
       if (codeModelGroup.operations?.length > 0) {
         codeModelClient.operationGroups.push(codeModelGroup);
       }
 
-      for (const operationGroup of operationGroups) {
-        const operations = listOperationsInOperationGroup(this.sdkContext, operationGroup);
+      // operations under operation groups
+      const subClients = this.listSubClientsUnderClient(client, true, true);
+      for (const subClient of subClients) {
+        const serviceMethods = this.listServiceMethodsUnderClient(subClient);
         // operation group with no operation is skipped
-        if (operations.length > 0) {
-          // groupPath would be in format of "[<SubNamespace>.]<ClientName>.Chat.Completions"
-          let operationGroupPath = operationGroup.groupPath;
-          if (clientSubNamespace && operationGroup.groupPath.startsWith(clientSubNamespace + ".")) {
-            // remove SubNamespace
-            operationGroupPath = operationGroupPath.slice((clientSubNamespace + ".").length);
-          }
-          const groupPath = operationGroupPath.split(".");
-          let operationGroupName: string;
-          if (groupPath.length > 1) {
-            // remove ClientName
-            operationGroupName = groupPath.slice(1).join("");
-          } else {
-            operationGroupName = groupPath[0];
-          }
-          codeModelGroup = new OperationGroup(operationGroupName);
-          for (const operation of operations) {
-            if (!this.needToSkipProcessingOperation(operation, clientContext)) {
-              codeModelGroup.addOperation(this.processOperation(operationGroupName, operation, clientContext));
+        if (serviceMethods.length > 0) {
+          codeModelGroup = new OperationGroup(subClient.name);
+          for (const serviceMethod of serviceMethods) {
+            if (!this.needToSkipProcessingOperation(serviceMethod.__raw, clientContext)) {
```

Comment:
Can we get rid of this raw?

Reply:
`needToSkipProcessingOperation` will check if the operation is overloaded or not by checking if the operation has `@overload` decorator, and TCGC does not have a flag indicating the operation is overload. So I use raw. I synced with Chenjie before and he said the overload method support is a big feature and needs many design discussions. I can confirm with him on whether there is plan and issue created, if no issue, I can help to create.

Reply:
Do we have an issue to track this in TCGC? If not, can we please create one?

Reply:
Yes, issue created https://github.com/Azure/typespec-azure/issues/1409